influxdb3_wal/src/snapshot_tracker.rs

```diff
@@ -118,6 +123,29 @@ impl SnapshotTracker {
         })
     }
 
+    fn snapshot_up_to_last_wal_period(&mut self) -> Option<SnapshotInfo> {
```

Comment:
This function name is a bit off. During normal snapshot operation, we want to snapshot the flush size number of wal periods. But we want to leave behind flush size / 2 periods. So that means we wait until we have flush size + flush size / 2 periods and then we take the oldest flush size periods, leaving behind flush size / 2.

If we're in a situation where we can't flush the WAL and leave behind periods because the time stamps of the data are all interleaved, we flush everything, except the most recent wal period. We do this because the buffer snapshots what is in it and then puts the most recent period into the snapshot.

In the case of forcing a snapshot, we don't need to check `should_snapshot`, we just treat it like the situation where we have 3x the flush size. So we want to snapshot everything minus the last wal period.